src/iter/normalize_reads.rs

```diff
@@ -0,0 +1,67 @@
+use crate::iter::*;
+
+pub struct NormalizeReads<R: Reads> {
+    reads: R,
+    selector_expr: SelectorExpr,
+    label: Label,
+    range: (usize, usize),
+}
+
+pub const NUC_MAP: [u8; 4] = [b'A', b'C', b'T', b'G'];
+
+pub fn log2_roundup(n: usize) -> usize {
```

Comment:
You can probably use [`leading_zeros`](https://stackoverflow.com/questions/72251467/computing-ceil-of-log2-in-rust)?